pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -118,11 +124,168 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 		// When CSIStorageCapacity is enabled, pods may become schedulable
 		// on CSI driver & storage capacity changes.
 		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}},
-		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterCSIStorageCapacityChange},
 	}
 	return events
 }
 
+func (pl *VolumeBinding) isSchedulableAfterStorageClassChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldStorageClass, modifiedStorageClass, err := util.As[*storagev1.StorageClass](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"StorageClass", klog.KObj(modifiedStorageClass),
+	)
+
+	if oldStorageClass.AllowVolumeExpansion != modifiedStorageClass.AllowVolumeExpansion {
+		logger.V(4).Info("StorageClass was created or updated, and changed AllowVolumeExpansion", "AllowVolumeExpansion", modifiedStorageClass.AllowVolumeExpansion)
+		return framework.Queue, nil
+	}
+
+	if oldStorageClass.VolumeBindingMode != modifiedStorageClass.VolumeBindingMode {
+		logger.V(4).Info("StorageClass was created or updated, and changed Provisioner", "VolumeBindingMode", modifiedStorageClass.VolumeBindingMode)
+		return framework.Queue, nil
+	}
+
+	logger.V(4).Info("StorageClass was created or updated, but it doesn't make this pod schedulable")
+	return framework.QueueSkip, nil
+}
+
+func (pl *VolumeBinding) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPVC, modifiedPVC, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"PersistentVolumeClaim", klog.KObj(modifiedPVC),
+	)
+
+	if pod.Namespace == modifiedPVC.Namespace {
+		for _, vol := range pod.Spec.Volumes {
+			var pvcName string
+			switch {
+			case vol.PersistentVolumeClaim != nil:
+				pvcName = vol.PersistentVolumeClaim.ClaimName
+			case vol.Ephemeral != nil:
+				pvcName = ephemeral.VolumeClaimName(pod, &vol)
+			}
+
+			if pvcName == modifiedPVC.Name {
+				if modifiedPVC.Status.Phase != oldPVC.Status.Phase {
```

Comment:
The fields of PVC referenced within Volumebinding are wide-ranging.
Therefore, if we implement a logic that strictly checks for changes in the fields being used, it would be difficult to correctly reflect those changes in the QueueingHintFn when there are modifications to PreFilter or Filter, and I think it would result in high maintenance costs.
So, for the Annotations and Spec fields, I'm using DeepEqual to check for differences. 
I would appreciate your thoughts on about this approach.